src/workers/parser/tests/contains.spec.js

```diff
@@ -0,0 +1,307 @@
+import {
+  containsPosition,
+  containsLocation,
+  nodeContainsPosition
+} from "../utils/contains";
+
+function getTestLoc() {
+  return {
+    start: {
+      line: 10,
+      column: 2
+    },
+    end: {
+      line: 12,
+      column: 10
+    }
+  };
+}
+
+function startPos(lineOffset, columnOffset) {
+  const { start } = getTestLoc();
+  return {
+    line: start.line + lineOffset,
+    column: start.column + columnOffset
+  };
+}
+
+function endPos(lineOffset, columnOffset) {
+  const { end } = getTestLoc();
+  return {
+    line: end.line + lineOffset,
+    column: end.column + columnOffset
+  };
+}
+
+function startLine(lineOffset = 0) {
+  const { start } = getTestLoc();
+  return {
+    line: start.line + lineOffset
+  };
+}
+
+function endLine(lineOffset = 0) {
+  const { end } = getTestLoc();
+  return {
+    line: end.line + lineOffset
+  };
+}
+
+function testContains(pos, bool) {
+  const loc = getTestLoc();
+  expect(containsPosition(loc, pos)).toEqual(bool);
+}
+
+function testContainsPosition(pos, bool) {
+  const loc = getTestLoc();
+  expect(nodeContainsPosition({ loc }, pos)).toEqual(bool);
+}
+
+describe("containsPosition", () => {
+  describe("location and postion both with the column criteria", () => {
+    it("should contain position within the location range", () =>
+      testContains(startPos(1, 1), true));
+
+    it("should not contain position out of the start line", () =>
+      testContains(startPos(-1, 0), false));
+
+    it("should not contain position out of the start column", () =>
+      testContains(startPos(0, -1), false));
+
+    it(`should contain position on the same start line and
+       within the start column`, () => testContains(startPos(0, 1), true));
+
+    it("should not contain position out of the end line", () =>
+      testContains(endPos(1, 0), false));
+
+    it("should not contain position out of the end column", () =>
+      testContains(endPos(0, 1), false));
+
+    it("should contain position on the same end line and within the end column", () =>
```

Comment:
@jasonLaster the eslint complains about a over-80-length line but looks like it was prettier which formatted this line...